test/DeviceStateTrackerSpec.ts

```diff
@@ -0,0 +1,140 @@
+				expect(deviceStateTracker.currentState).to.equal(DeviceState.X_SMALL);
+				deviceStateTracker.removeEventListener(DeviceStateEvent.STATE_UPDATE, eventHandler);
+			};
+
+			deviceStateTracker.addEventListener(DeviceStateEvent.STATE_UPDATE, eventHandler);
+
+			(<any> deviceStateTracker)._queryLists.forEach(mq => mq.callListeners());
+		});
+
+		it('should match SMALL', () => {
+			matchMediaMock.setConfig({ type: 'screen', width: 480 });
+
+			const eventHandler = () => {
+				expect(deviceStateTracker.currentState).to.equal(DeviceState.SMALL);
+				deviceStateTracker.removeEventListener(DeviceStateEvent.STATE_UPDATE, eventHandler);
+			};
+
+			deviceStateTracker.addEventListener(DeviceStateEvent.STATE_UPDATE, eventHandler);
+
+			(<any> deviceStateTracker)._queryLists.forEach(mq => mq.callListeners());
+		});
+
+		it('should match MEDIUM', () => {
+			matchMediaMock.setConfig({ type: 'screen', width: 768 });
+
+			const eventHandler = () => {
+				expect(deviceStateTracker.currentState).to.equal(DeviceState.MEDIUM);
+				deviceStateTracker.removeEventListener(DeviceStateEvent.STATE_UPDATE, eventHandler);
+			};
+
+			deviceStateTracker.addEventListener(DeviceStateEvent.STATE_UPDATE, eventHandler);
+
+			(<any> deviceStateTracker)._queryLists.forEach(mq => mq.callListeners());
+		});
+
+		it('should match LARGE', () => {
+			matchMediaMock.setConfig({ type: 'screen', width: 1280 });
+
+			const eventHandler = () => {
+				expect(deviceStateTracker.currentState).to.equal(DeviceState.LARGE);
+				deviceStateTracker.removeEventListener(DeviceStateEvent.STATE_UPDATE, eventHandler);
+			};
+
+			deviceStateTracker.addEventListener(DeviceStateEvent.STATE_UPDATE, eventHandler);
+
+			(<any> deviceStateTracker)._queryLists.forEach(mq => mq.callListeners());
+		});
+	});
+
+	it('should return the state-indicator div', () => {
+		const deviceStateTracker:DeviceStateTracker = new DeviceStateTracker(mediaQueries, DeviceState, false, true);
+		// Lint will throw an error if not defined (workaround)
+		if (deviceStateTracker) {
+			expect(document.querySelector('.seng-state-indicator')!.tagName).to.equal('DIV');
+		}
+	});
+
+	it('should throw an error while calling construct with wrong enum', () => {
+		expect(() => {
+			new DeviceStateTracker(mediaQueries, WrongDeviceState);
+		}).to.throw(
+			Error,
+		);
+	});
+
+	it('should throw an error when device state doesn\'t have any keys', () => {
+		expect(() => {
+			new DeviceStateTracker(mediaQueries, {});
+		}).to.throw(
+			Error,
+		);
+	});
+
+	it('should throw an error when device state isn\'t found in mediaQueries', () => {
+		const mediaQueriesClone = JSON.parse(JSON.stringify(mediaQueries));
+		delete mediaQueriesClone['MEDIUM'];
+
+		expect(() => {
+			new DeviceStateTracker(mediaQueriesClone, DeviceState);
+		}).to.throw(
+			Error,
+		);
+	});
+
+	it('should reverse the deviceState order', () => {
+		// Branch coverage
+		new DeviceStateTracker(mediaQueries, DeviceState, true);
```

Comment:
Nope doesn't test anything. I was thinking to reverse the array instead of looping using -1 but _queryList is still private so I'm not completly sure how to test it properly. 

What I could do though is making a second MediaQueryList and DeviceState that uses max-width to test it.